src/main/java/javax/measure/unit/UnitFormat.java

```diff
@@ -530,8 +534,31 @@ private void check(boolean expr, String message, CharSequence csq,
                         + " at index " + index + ")", index);
             }
         }
-        
-        private Exponent readExponent (CharSequence csq, ParsePosition pos) {
+
+        private void checkForBalancedParentheses(CharSequence csq) throws ParseException {
+            Stack<Character> stack = new Stack<Character>();
```

Comment:
Possible tweak: looks like you could use an int to count open-parens instead of a Stack